include/oneapi/dpl/pstl/utils.h

```diff
@@ -695,6 +695,26 @@ __iterators_possibly_equal(_Iterator1 __it1, _Iterator2 __it2)
     }
 }
 
+// Conditionally sets type to _SpirvT if oneDPL is being compiled to a SPIR-V target with the SYCL backend and _NonSpirvT otherwise.
+// This should be used only in kernels.
+template <typename _SpirvT, typename _NonSpirvT>
+struct __spirv_target_selector
+{
+    using type =
+#if _ONEDPL_DETECT_SPIRV_COMPILATION
+        _SpirvT;
+#else
+        _NonSpirvT;
+#endif
+};
+
+// Trait that has a true value if _ONEDPL_DETECT_SPIRV_COMPILATION is set and false otherwise. This may be used within kernels
+// to determine SPIR-V targets.
+struct __is_spirv_target : __spirv_target_selector<::std::true_type, ::std::false_type>
```

Comment:
By the way, I am not sure but perhaps we don't even need a `struct`. We might try 

```cpp
inline constexpr bool __is_spirv_target_v = __spirv_target_selector<std::true_type>::value;
```

Reply:
Yes, I do not think the struct has much use. I have made this change.